frontend/src/rauta.tsx

```diff
@@ -178,8 +203,8 @@ export const makeRouter = <C extends Config, >(config: C): RouterLib => {
     };
 
 
-    const shouldPreventNav = (listeners: { listener: BeforeNavigationListener }[]): boolean => {
-        for (const { listener } of listeners) {
+    const shouldPreventNav = (listeners: Listeners<BeforeNavListener>): boolean => {
+        for (const { listener } of listeners.list) {
```

Comment:
You could also make `Listeners` implement the iterator protocol :shrug:

Reply:
I think this is overkill. I've never done, but skimming the docs, it looks like not super trivial. So nah, it's just an internal helper anyway.

Reply:
Don't hate me, but I pushed a commit that does it. It's actually not that much overhead. If you hate it, we can remove it, but I wanted to try it. :sweat_smile: